template.go

```diff
@@ -195,18 +195,25 @@ func (t *template) parse(inputFile string) {
 	// Make the name mappings
 	t.newIsPublic = ast.IsExported(t.Name)
 
-	fset, f := parseFile(inputFile, nil)
-
-	conf := types.Config{Importer: importer.Default()}
-	info := &types.Info{
-		Defs: make(map[*ast.Ident]types.Object),
-		Uses: make(map[*ast.Ident]types.Object),
+	conf := &packages.Config{
+		Mode: packages.LoadSyntax,
 	}
-	_, err := conf.Check(inputFile, fset, []*ast.File{f}, info)
+
+	pkgs, err := packages.Load(conf, inputFile)
```

Comment:
Should this logic be factored into `parseFile` since there is another call to `parseFile` later on?

Reply:
From what I can tell, it doesn't look like go/packages supports parsing a source string as we do later on for header injection. Since the main issue was in conf.Check, it seems ok to leave the latter just for parsing without checks (as we already do it here).

Reply:
That sounds fine - thanks for explaining :-)